services/data-record-service.js

```diff
@@ -96,6 +120,59 @@ class Message {
     }
 }
 
+class Stat {
+    constructor(nodeName, totalCount, newCount, passedCount, warningCount, errorCount) {
+        this.nodeName = nodeName;
+        this.total = totalCount;
+        this.new = newCount;
+        this.passed = passedCount;
+        this.warning= warningCount;
+        this.error = errorCount;
+    }
+
+    static createStat(nodeName) {
+        return new Stat(nodeName, 0,0,0,0, 0);
+    }
+
+    #addTotal(total) {
```

Comment:
I think we should remove the "#" from the start of this function name. I don't know if there is any issue with it but I'd rather keep the function names alphanumeric to match the existing code style. 

If there is a special function for this naming convention please let me know

Reply:
It is the purpose of private method